### PR TITLE
Add Advanced Cache Upsell

### LINKED ISSUE #41119: Put Cache Upsell component on OSS page

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/41119

see [discussion](https://metaboat.slack.com/archives/C06KX7QECN4/p1716475479960519)

### Description

- Adds a `useHasTokenFeature` hook
- Adds an upsell for advanced caching for admins that don't  have it
- Displays that upsell on the performance tab

Screen Shot 2024-05-24 at 7 50 39 AM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by mazameli — CHANGES_REQUESTED** [review]
The upsell card component [should have a white bg](https://metaboat.slack.com/archives/C02H619CJ8K/p1716509278450759?thread_ts=1716509208.660149&cid=C02H619CJ8K).

**Comment by iethree** [comment]
> The upsell card component [should have a white bg](https://metaboat.slack.com/archives/C02H619CJ8K/p1716509278450759?thread_ts=1716509208.660149&cid=C02H619CJ8K).

thanks for catching that. fixed!
Screen Shot 2024-05-24 at 7 50 39 AM

**Review by luizarakaki — APPROVED** [review]

**Comment by mazameli** [comment]
Dunno if this is problematic, but /admin/performance exploded when I accidentally built the FE and ran the server via `yarn dev-ee` instead of via `yarn dev`:

image

```
[backend] 2024-05-29 10:21:36,101 INFO i18n.impl :: Reading available locales from locales.clj...
[backend] 2024-05-29 10:21:36,150 INFO util.fonts :: Reading available fonts from /Users/maz/workspace/metabase/resources/frontend_client/app/fonts
[backend] 2024-05-29 10:21:39,551 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:21:39,807 DEBUG middleware.log :: GET /api/session/properties 200 651.7 ms (10 DB calls) App DB connections: 0/4 Jetty threads: 6/50 (1 idle, 0 queued) (45 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:39,934 DEBUG middleware.log :: GET /api/user/current 200 796.7 ms (13 DB calls) App DB connections: 0/4 Jetty threads: 6/50 (1 idle, 0 queued) (45 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,320 DEBUG middleware.log :: GET /api/database 200 255.8 ms (1 DB calls) App DB connections: 0/7 Jetty threads: 9/50 (1 idle, 0 queued) (48 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,478 DEBUG middleware.log :: GET /api/collection/root 200 413.1 ms (2 DB calls) App DB connections: 1/7 Jetty threads: 8/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,515 DEBUG middleware.log :: GET /api/bookmark 200 308.2 ms (1 DB calls) App DB connections: 1/7 Jetty threads: 7/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,676 DEBUG middleware.log :: GET /api/collection/tree 200 419.9 ms (6 DB calls) App DB connections: 2/7 Jetty threads: 10/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,858 DEBUG middleware.log :: GET /api/util/bug_report_details 200 187.6 ms (1 DB calls) App DB connections: 4/7 Jetty threads: 10/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:40,891 DEBUG middleware.log :: GET /api/collection/trash 200 218.0 ms (4 DB calls) App DB connections: 3/7 Jetty threads: 10/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:41,187 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,205 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,215 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,224 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,227 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,228 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,229 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,232 ERROR models.recent-views :: Invalid recent view item: nil reason: ["invalid type" "invalid dispatch value"]
[backend] 2024-05-29 10:21:41,238 DEBUG middleware.log :: GET /api/activity/recent_views 200 608.0 ms (7 DB calls) App DB connections: 2/7 Jetty threads: 8/50 (1 idle, 0 queued) (41 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:41,411 DEBUG middleware.log :: GET /api/activity/popular_items 200 780.3 ms (10 DB calls) App DB connections: 0/7 Jetty threads: 7/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:41,629 DEBUG middleware.log :: GET /api/search 200 1.4 s (6 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:21:42,251 DEBUG middleware.log :: GET /api/automagic-dashboards/database/2/candidates 200 1.2 s (7 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (4 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:00,443 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:00,580 DEBUG middleware.log :: GET /api/session/properties 200 224.9 ms (10 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:00,694 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:00,800 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:00,828 DEBUG middleware.log :: GET /api/setting 200 469.5 ms (25 DB calls) App DB connections: 1/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:00,855 DEBUG middleware.log :: GET /api/session/properties 200 221.3 ms (10 DB calls) App DB connections: 0/7 Jetty threads: 7/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:01,150 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:01,170 DEBUG middleware.log :: GET /api/setup/admin_checklist 200 186.7 ms (11 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:01,170 DEBUG middleware.log :: GET /api/setting 200 321.3 ms (25 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:01,408 DEBUG middleware.log :: GET /api/cache 200 33.0 ms (1 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:01,408 DEBUG middleware.log :: GET /api/cache 200 32.8 ms (1 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:04,967 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:05,052 DEBUG middleware.log :: GET /api/user/current 200 129.4 ms (11 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (5 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,240 DEBUG middleware.log :: GET /api/session/properties 200 310.1 ms (10 DB calls) App DB connections: 3/7 Jetty threads: 10/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,294 DEBUG middleware.log :: GET /api/cache 200 49.5 ms (1 DB calls) App DB connections: 1/7 Jetty threads: 10/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,295 DEBUG middleware.log :: GET /api/cache 200 52.0 ms (1 DB calls) App DB connections: 1/7 Jetty threads: 10/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,348 DEBUG middleware.log :: GET /api/database 200 114.2 ms (1 DB calls) App DB connections: 0/7 Jetty threads: 9/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,553 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:05,791 DEBUG middleware.log :: GET /api/session/properties 200 390.9 ms (10 DB calls) App DB connections: 0/7 Jetty threads: 7/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,933 WARN metabase.public-settings :: 'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead
[backend] 2024-05-29 10:22:05,952 DEBUG middleware.log :: GET /api/setting 200 721.9 ms (25 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[backend] 2024-05-29 10:22:05,995 DEBUG middleware.log :: GET /api/search 200 628.7 ms (5 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (3 idle, 0 queued) (43 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
```

**Comment by iethree** [comment]
> Dunno if this is problematic, but /admin/performance exploded when I accidentally built the FE and ran the server via `yarn dev-ee` instead of via `yarn dev`:

👍  had a bad rebase. tests caught it too.

**Review by mazameli — APPROVED** [review]

**Review by sloansparger — APPROVED** [review]
Nice! I like the addition of `useHasTokenFeature`.

File: frontend/src/metabase/admin/upsells/components/Upsells.styled.tsx [hunk 1]
```diff
@@ -64,4 +64,5 @@ export const UpsellCardComponent = styled.div`
   border-radius: 0.5rem;
   overflow: hidden;
   border: 1px solid ${upsellColors.secondary};
+  background-color: ${color("white")};
```
Comment (sloansparger) [inline frontend/src/metabase/admin/upsells/components/Upsells.styled.tsx:67]: why not just "white"?

**Pull request merged**